blueocean-dashboard/src/main/js/components/PipelinePage.jsx

```diff
@@ -20,6 +20,8 @@ export default class PipelinePage extends Component {
             return null; // Loading...
         }
 
+        const baseUrl = removeLastUrlSegment(this.context.location.pathname);
```

Comment:
Is there a reason we're constructing a URL by chopping up what's in the browser's state, rather than the app's base URL? It makes it harder to figure out the URL by reading the code, and will almost certainly break as soon as the URLs get deeper because somebody has changed routes further down the component/route tree.